### PR TITLE
Add User Mgmt & Sharing docs

### PR DESCRIPTION
**Pull request by htroisi**
Signed-off-by: Hannah Troisi <htroisi@pixielabs.ai>

### CODE REVIEW

File: content/en/05-reference/01-admin/01-authentication.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ redirect_from:
 
 ## Open Source Auth
 
-Pixie uses [Hydra](https://www.ory.sh/hydra/)/[Kratos](https://www.ory.sh/kratos/) to provide an  open source authentication flow.
+[Self-Hosted Pixie](/installing-pixie/install-guides/self-hosted-pixie/) uses [Hydra](https://www.ory.sh/hydra/)/[Kratos](https://www.ory.sh/kratos/) to provide an open source authentication flow.
```
Comment (vihangm) [inline content/en/05-reference/01-admin/01-authentication.md]: This seems like an unnecessary change. We are already in the OS Auth section.

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+<svg title='' src='admin/user-mgmt/users-tab.png'/>
+
+4. Accept the prompt confirming the removal.
+
+<svg title='' src='admin/user-mgmt/user-removal-prompt.png'/>
+
+Once a user is removed, they will be automatically logged out from any active UI sessions and will be unable to query Pixie resources.
+
+If you mistakenly remove a user and need to add the user back, you can create a new invite link and send it to the user. The user will be able to use their existing credentials to log in and join your org.
```
Comment (vihangm) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: should we drop this? because the general caveat of GSuite user vs not also applies here and people are liable to read bits of this page without reading other sections.

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+
+<svg title='' src='admin/user-mgmt/enable-approvals.png'/>
+
+4. Manually approve new users under the Users tab (`/admin/org`).
+
+>> Once User Approvals are enabled, any new users that join your org will join in a disabled state.
+
+<svg title='' src='admin/user-mgmt/user-unapproved.png'/>
+
+>> Click “Approve” on the appropriate user to allow that user complete access to your organization within Pixie.
```
Comment (vihangm) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: Indenting this text seems strange, but I'm sure it's strange to not indent it either.

**Comment by vihangm** [comment]
Mostly LGTM! Couple of quick comments.

File: content/en/05-reference/01-admin/09-user-mgmt.md (hunk 3 above)
Comment (htroisi) [reply content/en/05-reference/01-admin/09-user-mgmt.md]: It looks worse without IMO.

File: content/en/05-reference/01-admin/01-authentication.md [hunk 4]
```diff
@@ -9,7 +9,7 @@ redirect_from:
 
 ## Open Source Auth
 
-[Self-Hosted Pixie](/installing-pixie/install-guides/self-hosted-pixie/) uses [Hydra](https://www.ory.sh/hydra/)/[Kratos](https://www.ory.sh/kratos/) to provide an open source authentication flow.
+Pixie uses [Hydra](https://www.ory.sh/hydra/)/[Kratos](https://www.ory.sh/kratos/) to provide an  open source authentication flow.
```
Comment (vihangm) [inline content/en/05-reference/01-admin/01-authentication.md]: accidental double space between `an` and `open`

**Review by vihangm — DISMISSED** [review]

**Review by nserrino — COMMENTED** [review]
image
can we alphabetize these?

**Comment by nserrino** [comment]
> image can we alphabetize these?

image
 here as well

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 5]
```diff
@@ -0,0 +1,107 @@
+---
+title: "User Management & Sharing"
+metaTitle: "Reference | Admin | User Management & Sharing"
+metaDescription: "Share and manage access to your Pixie data."
+order: 6
+redirect_from:
+    - /admin/permissions/
+---
+
+<Alert variant="outlined" severity="info">This information applies to Pixie instances that use Auth0 for Authentication. <a href="https://docs.px.dev/installing-pixie/install-guides/community-cloud-for-pixie/">Community Cloud for Pixie</a> ships with Auth0. <a href="https://docs.px.dev/installing-pixie/install-guides/self-hosted-pixie/">Self-Hosted Pixie</a> must be <a href="https://docs.px.dev/reference/admin/authentication/#enabling-auth0">manually configured</a> to use Auth0. To learn how to invite users to Self-Hosted Pixie instances using open source authentication, see the Quick Start <a href="https://docs.px.dev/installing-pixie/install-guides/self-hosted-pixie/#invite-others-to-your-organization-(optional)">instructions</a>.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: it seems a little odd to me to have the invite instructions for non Auth0 OSS in the install instructions and the invite instructions for Auth0 OSS in this page. Can we put them in here in both cases, and also link from all sets of the install instructions to this page? I think we will want to say more about user management & sharing in non-auth0 over time as we add more features

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+Pixie handles GSuite orgs differently than other orgs.
+
+You are part of a Pixie GSuite org if:
+
+1. The original creator of the org signs into Pixie using the "Sign in with Google" button AND
+2. The email the admin uses to sign in is part of a GSuite account.
+
+GSuite orgs can invite two types of users:
+
+- **Teammates with GSuite emails**: should use the "Sign-Up with Google" button on the [Sign up](https://work.withpixie.ai/auth/signup) page. GSuite emails will automatically join the Pixie org corresponding to their GSuite domain. _DO NOT use Invite Links to invite teammates with GSuite emails._ For greater control, enable [User Approvals](/reference/admin/user-mgmt#enable-approvals-for-new-users).
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: > DO NOT use Invite Links to invite teammates with GSuite emails.

@aimichelle can you weigh in on this - it seems like it will be an easy mistake for users to make. I wonder if it makes sense to augment the flow to prevent people from falling into pits.

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+- **Teammates using Gmail or creating custom email/password Pixie accounts**: should be sent [Invite Links](/reference/admin/user-mgmt#invite-links).
+
+### All Other Organizations
+
+If your organization is not a [GSuite organization](/reference/admin/user-mgmt#gsuite-organnizations), you can invite users with [Invite Links](/reference/admin/user-mgmt#invite-links).
+
+<Alert variant="outlined" severity="warning">Invited teammates using GSuite emails MUST sign up for Pixie using the "Sign-Up with Email" button. If a teammate accidentally signs up for Pixie using SSO ("Sign-Up with Google" button), they will not be able to join your Org. To fix this, they will need to sign up for Pixie again using the "Sign-Up with Email" button with a new or modified email (e.g. adding `+pixie` to their email, like anna+pixie@fakeemail.com)</Alert>
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: @aimichelle  - I understand that we have these requirements for good reason and not challenging the requirements themselves, but I wonder if there is a way to augment the flows to add guardrails/etc during the invite link or signup process to make falling into the "bad" cases less easy to do

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+
+### Invite Links
+
+1. Open the [Live UI](/using-pixie/using-live-ui) Admin page (`/admin`).
+2. Select the Users tab (`/admin/org`).
+
+<svg title='' src='admin/user-mgmt/users-tab-invite-button.png'/>
+
+3. Click the “+ Invite Users” button in the top right corner. This will pop open a modal with a generated invite link for your organization.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: nit: add backticks for `+ Invite Users` so it pops

File: content/en/05-reference/01-admin/09-user-mgmt.md [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+
+<svg title='' src='admin/user-mgmt/invite-link.png'/>
+
+4. Copy the link and send it to your teammates.
+
+### Enable Approvals for New Users
+
+Combine Invite Links with User Approvals for greater control over who joins your organization. If your organization enables User Approvals, every user must be manually approved after they join your organization before they are able use Pixie.
```
Comment (nserrino) [inline content/en/05-reference/01-admin/09-user-mgmt.md]: > Combine Invite Links with User Approvals for greater control over who joins your organization.

Slightly confused here - Invite links are NOT for GSuite users. But I thought user approvals could be. If user approval are handled differently from gsuite to username/password users, then we may want to just have a section for each type of account, and each subsection (Such as enable approvals) repeated within each section.

**Review by nserrino — CHANGES_REQUESTED** [review]
Overall, this needs to be changed to make it clear exactly what applies to each bucket of users.

From a docs perspective, I had confusion about the following groups of users:
- Non-Auth0 users - does any of this apply to them?
- GSuite users - they can't use invite links, but can they use the other features?

From a project perspective, I'm interested to see if there's an opportunity to reduce the odds that users make mistakes with these flows, even while preserving our org requirements.

File: content/en/05-reference/01-admin/09-user-mgmt.md (hunk 9 above)
Comment (htroisi) [reply content/en/05-reference/01-admin/09-user-mgmt.md]: IIUC, User Approvals also applies to Gsuite users (who join orgs automatically). I think the confusion could be resolved if I just delete the first line here? @vihangm

Comment (vihangm) [reply content/en/05-reference/01-admin/09-user-mgmt.md]: GSuite based orgs can also invite users (the non GSuite kind). Invites scoping is at the org level so is user approval which is why it was phrased this way.
You are right that approvals will gate all users. Invited or auto added using GSuite. This was just written in a way to suggest that you have extra control over invite links by leveraging Approvals.
Feel free to reword.

File: content/en/05-reference/01-admin/09-user-mgmt.md (hunk 6 above)
Comment (vihangm) [reply content/en/05-reference/01-admin/09-user-mgmt.md]: This will already gate users and return an error message of the form 
`gsuite users are not allowed to follow invites. Please join the org with another account`
We can improve the error and add some more safeguards on our side as need be.

cc @philkuz

Comment (vihangm) [reply content/en/05-reference/01-admin/09-user-mgmt.md]: A quick improvement we could do is to check if the GSuite domain matches the domain for the invite link they followed and let the user continue in that scenario (because that's the org the user would be in without and invite if they just did Google SSO).

File: content/en/05-reference/01-admin/11-user-mgmt.md [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+
+1. Navigate to [`dev.withpixie.dev/admin/org`](https://dev.withpixie.dev/admin/org) in your browser.
+
+2. Click `“Enable”` on the Approvals setting to enable Approvals. Note that if the button says `“Disable”`, this means User Approvals has already been enabled for your organization.
+
+<svg title='' src='admin/user-mgmt/os-enable-approvals.png'/>
+
+3. Manually approve new users under the Users tab ([`dev.withpixie.dev/admin/users`](https://dev.withpixie.dev/admin/users)). Once User Approvals is enabled, any new users that join your org will join in a disabled state. Click `“Approve”` on the appropriate user to allow that user complete access to your organization within Pixie.
+
+### Remove Users
```
Comment (aimichelle) [inline content/en/05-reference/01-admin/11-user-mgmt.md]: After discussing with Phillip & Vihang, it looks like enabling this for Hydra/Kratos isn't as simple as we originally expected... Removed users can currently end up in a bad state, so we won't be enabling this for Hydra/Kratos in the meantime. Lets remove this section.

**Review by aimichelle — DISMISSED** [review]
Just one comment, otherwise looks good to me.

**Review by aimichelle — DISMISSED** [review]

**Review by nserrino — APPROVED** [review]
